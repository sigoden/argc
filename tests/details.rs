use super::SCRIPT_DETAILS;

#[test]
fn wrap() {
    snapshot!(SCRIPT_DETAILS, &["prog", "-h"], None, Some(80));
}

#[test]
fn wrap2() {
    snapshot!(SCRIPT_DETAILS, &["prog", "foo", "-h"], None, Some(80));
}

#[test]
fn nowrap() {
    snapshot!(SCRIPT_DETAILS, &["prog", "-h"], None, None);
}
