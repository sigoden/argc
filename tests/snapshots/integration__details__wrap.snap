---
source: tests/details.rs
expression: data
---
RUN
prog -h

# OUTPUT
command cat >&2 <<-'EOF' 
Multi-line auto-wrapped help text

Extra lines after the @cmd or @describe, which don't start with an @, are
treated as the long description. A line which is not a comment ends
the block.

USAGE: prog [OPTIONS] [TARGET] <COMMAND>

ARGS:
  [TARGET]  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua. Sed
            viverra tellus in hac habitasse platea.
            Use '-' for standard input.
            


OPTIONS:
      --foo <FOO>  Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                   sed do eiusmod tempor incididunt ut labore et dolore magna
                   aliqua. Neque laoreet suspendisse libero id.
                    * default: enables recommended style components.
                    * full: enables all available components.
                    * auto: same as 'default', unless the output is piped.
                   
                   [default: default]
                   [possible values: default, full, auto]

      --bar <BAR>  Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                   sed do eiusmod tempor incididunt ut labore et dolore magna
                   aliqua. Neque laoreet suspendisse libero id.

  -h, --help       Print help

  -V, --version    Print version


COMMANDS:
  cmd  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
       tempor incididunt ut labore et dolore magna aliqua. Arcu cursus euismod
       quis viverra.


EOF
exit 0

# BUILD_OUTPUT
Multi-line auto-wrapped help text

Extra lines after the @cmd or @describe, which don't start with an @, are
treated as the long description. A line which is not a comment ends
the block.

USAGE: prog [OPTIONS] [TARGET] <COMMAND>

ARGS:
  [TARGET]  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Sed viverra tellus in hac habitasse platea.
            Use '-' for standard input.
            


OPTIONS:
      --foo <FOO>  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Neque laoreet suspendisse libero id.
                    * default: enables recommended style components.
                    * full: enables all available components.
                    * auto: same as 'default', unless the output is piped.
                   
                   [default: default]
                   [possible values: default, full, auto]

      --bar <BAR>  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Neque laoreet suspendisse libero id.

  -h, --help       Print help

  -V, --version    Print version


COMMANDS:
  cmd  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Arcu cursus euismod quis viverra.


